src/lib.rs

```diff
@@ -1734,9 +1749,51 @@ impl Connection {
 
         let mut ack_eliciting = false;
         let mut in_flight = false;
+        let mut pmtud = false;
 
         let mut payload_len = 0;
 
+        // Create PMTUD probe.
+        //
+        // In order to send a PMTUD probe the current `left` value, which was
+        // already limited by the current PMTU measure, needs to be ignored, but
+        // the outgoing packet still needs to be limited by the output buffer
+        // size, as well as the congestion window.
+        //
+        // In addition, the PMTUD probe is only generated when the handshake is
+        // confirmed, to avoid interfering with the handshake (e.g. due to the
+        // anti-amplification limits).
+        if pkt_type == packet::Type::Short &&
+            (self.handshake_confirmed || self.handshake_done_sent) &&
+            self.pmtu_next > self.pmtu &&
+            self.recovery.cwnd_available() > self.pmtu_next &&
+            out_len > self.pmtu_next
+        {
+            // Add PADDING up to the PMTUD probe size (accounting for the QUIC
```

Comment:
Would it be useful to add some trace-level logging to indicate that a probe is being sent out?